builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java

```diff
@@ -78,7 +80,7 @@ public void visit(String attrName,
         }
 
         Object val = valueSupplier.get();
-        if (Objects.nonNull(val)) {
+        if (val != null && val != ConfiguredOption.UNCONFIGURED) {
```

Comment:
This is wrong. You cannot use instance comparison here.
Should be `if (val != null && !ConfiguredOption.UNCONFIGURED.equals(val))`

Reply:
pushed